src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs

```diff
@@ -125,14 +125,14 @@ private static ObjectType SetBicepResourceProperties(ObjectType objectType, Reso
                 if (validParentScopes.HasFlag(ResourceScope.Resource))
                 {
                     var scopeReference = LanguageConstants.CreateResourceScopeReference(ResourceScope.Resource);
-                    properties = properties.SetItem(LanguageConstants.ResourceScopePropertyName, new TypeProperty(LanguageConstants.ResourceScopePropertyName, scopeReference, scopeRequiredFlag));
+                    properties = properties.SetItem(LanguageConstants.ResourceScopePropertyName, new TypeProperty(LanguageConstants.ResourceScopePropertyName, scopeReference, scopePropertyFlags));
                 }
             }
             // Deployments RP
             if (StringComparer.OrdinalIgnoreCase.Equals(objectType.Name, TemplateWriter.NestedDeploymentResourceType))
             {
-                properties = properties.SetItem("resourceGroup", new TypeProperty("resourceGroup", LanguageConstants.String));
-                properties = properties.SetItem("subscriptionId", new TypeProperty("subscriptionId", LanguageConstants.String));
+                properties = properties.SetItem("resourceGroup", new TypeProperty("resourceGroup", LanguageConstants.String, TypePropertyFlags.DeployTimeConstant));
```

Comment:
>DeployTimeConstant [](start = 143, length = 18)

Do we allow accessing/reading these via property access? And do we codegen that correctly?

Reply:
Yes, Tarun added this recently. We just pass thru the properties to codegen the same we we handle other top-level properties.

Reply:
Good catch, thanks for this!